### ISSUE TITLE
Support the use of GitHub markdown alerts for the specs

### ISSUE THREAD

**Issue by chalin** [open-telemetry/opentelemetry.io#5557]
GitHub markdown alerts are now being used in the OTel spec. I would be a nice to have to support their proper formatting (which I thought that Hugo already did -- I'll have to investigate, it might require a bit of extra coding).

For example, see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#logs-api.

https://github.com/open-telemetry/opentelemetry-specification/blob/8bc96330a1c17ff5194c5afcf7d32b9ee24b65e3/specification/logs/sdk.md?plain=1#L542-L545:

```markdown
## Logs API


> [!NOTE]
> We are currently in the process of defining a new [Logs API](./api.md#logs-api).
```

This currently renders as follows for the OTel website:

> <img width="531" alt="image" src="https://github.com/user-attachments/assets/67a945c9-6250-4e8d-b4d2-5701e3712458">

On GitHub, it looks like this (we don't have to match the look exactly):

> <img width="467" alt="image" src="https://github.com/user-attachments/assets/92bfc872-d853-4a0d-9cab-9c56682b6853">

We might want to consider wrapping that into an `{{% alert %}}`?

**Comment by svrnm** [comment]
When you say wrapping that into the alert shortcode do you mean we write some transformation code that turns

```
> [!NOTE]
> We are currently in the process of defining a new [Logs API](./api.md#logs-api).
```

into

```
{{% alert title="Note" color="info" %}}
We are currently in the process of defining a new [Logs API](./api.md#logs-api).
{{% /alert %}}
```

?

**Comment by chalin** [comment]
Yes. The idea is that Hugo would be doing that automatically for us using a block-quote hook.

**Comment by svrnm** [comment]
this is still pending: https://github.com/open-telemetry/opentelemetry.io/issues/7464

**Comment by svrnm** [comment]
there seems to be a hugo upstream feature: 

https://gohugo.io/render-hooks/blockquotes/#alerts

**Comment by vitorvasc** [comment]
The Logs SDK page was updated, and that example is no longer there. 

Here's another example from a Semantic Conventions page that still uses it: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#status

https://opentelemetry.io/docs/specs/semconv/http/http-spans/#status

<img width="1462" height="416" alt="Image" src="https://github.com/user-attachments/assets/42bf10b0-1d84-452d-989f-8704e9f1e765" />

**Comment by chalin** [comment]
@svrnm @vitorvasc et all - Question: do we want this special processing to be active only for spec pages, or across the site? (I'm ok either way.)

**Comment by vitorvasc** [comment]
@chalin I like the idea of having it across the site 😄

**Comment by tiffany76** [comment]
I agree with @vitorvasc, as I think the issue might apply to some of the Collector component docs...when we ever get around to deciding what to do with them.

**Comment by chalin** [comment]
We might add some general guidance to the contrib pages to say which format we wish to give preference to. Do we want to encourage contributors to use the new alert syntax?

**Comment by vitorvasc** [comment]
> We might add some general guidance to the contrib pages to say which format we wish to give preference to. Do we want to encourage contributors to use the new alert syntax?

I see it as an "upgrade" if compared to the older (or default?) alert format, so yes, I think it makes sense to encourage contributors to use the new syntax.

The same thought applies to #8479 as well, encourage the usage of emojis once we have them enabled (as long its usage is  moderate and consistent, of course).
